Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1996,6 +1998,14 @@ extension BrowserViewController: ContextMenuHelperDelegate {
             popoverPresentationController.permittedArrowDirections = .Any
         }
 
+        if (count(dialogTitle!) >= maxDialogUrlLength) {
+            let unwrappedDialog = dialogTitle!
+            var builderString = unwrappedDialog[unwrappedDialog.startIndex..<advance(unwrappedDialog.startIndex, maxDialogUrlLength / 2)]
+            builderString.splice(unwrappedDialog[advance(unwrappedDialog.endIndex, maxDialogUrlLength / -2)..<unwrappedDialog.endIndex], atIndex: builderString.endIndex)
+            builderString.splice("...", atIndex: advance(unwrappedDialog.startIndex, maxDialogUrlLength / 2))
```

Comment:
Use a real Unicode ellipsis: "…". Which should make your calculations a little simpler, because you're splicing a single character.